rosetta/services/block.go

```diff
@@ -341,6 +538,48 @@ func formatGenesisTransaction(
 	return nil, &common.TransactionNotFoundError
 }
 
+// formatPreStakingBlockRewardsTransaction for block rewards in pre-staking era for a given Bech-32 address
+func formatPreStakingBlockRewardsTransaction(
+	b32Address string, blockSigInfo *blockSignerInfo,
+) (*types.Transaction, *types.Error) {
+	addr, err := internalCommon.Bech32ToAddress(b32Address)
+	if err != nil {
+		return nil, common.NewError(common.CatchAllError, map[string]interface{}{
+			"message": err.Error(),
+		})
+	}
+
+	signatures, ok := blockSigInfo.committee[addr]
+	if !ok || len(signatures) == 0 {
+		return nil, &common.TransactionNotFoundError
+	}
+	accID, rosettaError := newAccountIdentifier(addr)
+	if rosettaError != nil {
+		return nil, rosettaError
+	}
+
+	rewardsPerSlot := new(big.Int).Quo(stakingNetwork.BlockReward, big.NewInt(int64(blockSigInfo.totalKeysSigned)))
```

Comment:
![image](https://user-images.githubusercontent.com/1042540/91494063-1cc56800-e86d-11ea-82fd-b81ce2157080.png)
Better follow this exact logic for reward calculation, otherwise, there can be precision issues.

Reply:
Sure, I updated the calculation logic to follow this